packages/block/src/header.ts

```diff
@@ -165,6 +178,25 @@ export class BlockHeader {
     return BlockHeader.fromHeaderData(headerData, opts)
   }
 
+  public static getCommon(opts: BlockOptions = {}, blockNumber: BN = ZERO_BN): Common {
+    let common
+    if (opts.common) {
+      common = opts.common.copy()
+    } else {
+      const chain = 'mainnet' // default
+      if (opts.initWithGenesisHeader) {
+        common = new Common({ chain, hardfork: 'chainstart' })
+      } else {
+        // This initializes on the Common default hardfork
+        common = new Common({ chain })
+      }
+    }
+    if (opts.hardforkByBlockNumber) {
+      common.setHardforkByBlockNumber(blockNumber)
+    }
+    return common
+  }
```

Comment:
This is a bit a strange function to have publicly exposed, I wouldn't want to do this - at least not without prior additional discussion - I currently not see the direct use case respectively this would lead to strange patterns like:

```typescript
const common = new Common()
const common2 = BlockHeader.getCommon({ common })
const blockHeader = BlockHeader.fromHeaderData({}, { common2 })
``` 

At least my fear is that people will do these kind of things. Will switch to an implementation where this is private.

Reply:
Ok, this a199138 should very much dot it without changing the semantics too much and without the need for this `getCommon()` extraction, sorry if this feels a bit like a roll-back of work.

We should really really urgently follow the pattern from tx in `Block` and do a centralized default value setting, passing in values as a dict in the constructor instead of the values just one-by-one (this is already noted for v6) and then just having `BlockHeader.fromHeaderData()` as an alias to the constructor. The problems here with the need for having the final common in the static methods comes at the end from this defragmentation and doing fragmented checks in the various static constructors which should just be done once in the main constructor.

Anyhow: please note that this change also introduces a more explicit default value for `baseFeePerGas`, before this was somewhat implicitly set to `0` with `opts.common.isActivatedEIP(1559) ? new BN(toBuffer(baseFeePerGas)) : undefined`. I now have set this to `1`, but totally unsure here, what would actually be a good default value here? 🤔